fixing Databricks detector

4390c336 opened the pull request:
I'm just sending our internal developed detector

### Description:
we used this detector at Databricks for couple of months, 
and I noticed that the current detector is broken in many ways , so I just copy/pasted a working code.

![image](https://github.com/trufflesecurity/trufflehog/assets/63428291/5df4e04c-cc1d-4b3e-ae72-ea16246a6632)

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/trufflesecurity/trufflehog?pullRequest=1986) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/trufflesecurity/trufflehog?pullRequest=1986) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/trufflesecurity/trufflehog?pullRequest=1986) it.</sub>

4390c336 commented:
the CLA check is not working. I singed the doc multiple times

4390c336 commented:
@ahrav I didn't submit the PR for the competition, 
I just want this detector to be working so I don't have to build the code from source.

rosecodym commented on pkg/detectors/databrickstoken/databrickstoken.go:
```diff
@@ -21,52 +21,54 @@ var (
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	domain = regexp.MustCompile(`\b(https:\/\/[a-z0-9-]+\.cloud\.databricks\.com)\b`)
-	keyPat = regexp.MustCompile(`\b(dapi[a-z0-9]{32})\b`)
+	keyPat    = regexp.MustCompile(`\b(dapi[0-9Aa-f]{32})\b`)
```
Are you sure this pattern is correct? `[0-9Aa-f]` is an unusual character set. I would expect either `[0-9A-Fa-f]` or `[0-9a-f]`.

4390c336 replied:
it should be `[0-9a-f]` that was a mistake, thank you for spotting that. I will fix it right away

rosecodym commented on pkg/detectors/databrickstoken/databrickstoken.go:
```diff
@@ -21,52 +21,54 @@ var (
 
-	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
-	domainMatches := domain.FindAllStringSubmatch(dataStr, -1)
+	matches := keyPat.FindAll(data, -1)
+	domainMatches := domainPat.FindAll(data, -1)
```
If you use `FindAllString` instead of `FindAll` you won't have to manually convert back to `string` below.

rosecodym commented on pkg/detectors/databrickstoken/databrickstoken.go:
```diff
@@ -21,52 +21,54 @@ var (
 			s1 := detectors.Result{
 				DetectorType: detectorspb.DetectorType_DatabricksToken,
 				Raw:          []byte(resMatch),
-				RawV2:        []byte(resMatch + resDomainMatch),
```
Why did you remove `RawV2`?

4390c336 replied:
I wrote this detector long-time ago , not sure if RawV2 was there. I will add it. and test the code again

rosecodym reviewed (commented):
Thank you for taking a look at this detector! We'd love to get some fixes in but I have some questions about the specifics.

4390c336 commented:
I updated the code based on the suggestions.
I also fixed a missing case with the regex where the token sometimes have an extra -\d at the end.

and I also submitted my _test.go since the existing one didn't expect a domain

zricethezav commented on pkg/detectors/databrickstoken/databrickstoken.go:
```diff
@@ -21,64 +20,66 @@ var (
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	domain = regexp.MustCompile(`\b(https:\/\/[a-z0-9-]+\.cloud\.databricks\.com)\b`)
-	keyPat = regexp.MustCompile(`\b(dapi[a-z0-9]{32})\b`)
+	keyPat    = regexp.MustCompile(`\b(dapi[0-9a-f]{32})(-\d)?\b`)
```
@4390c336 we're gonna prioritize merging https://github.com/trufflesecurity/trufflehog/pull/1965/files since it was opened first. The most important change is expanding the `domain` which both PRs include. However, I'm curious about this change to `keyPat`. Could you explain the reason behind this change? Do all databricks tokens have an appened `-{digit}`?

cc @ankushgoel27 (you don't need to do anything atm besides respond to the changes I requested in your PR, but wanted to make you aware of this one too)

4390c336 replied:
@zricethezav as indicated by regex . the `-{digit}` part is optional. not all tokens have it. some of them do.

source : I work for Databricks, and we have this running to scan our internal/public repos.

4390c336 replied:
PS : i don't care about the competition or credit as long as we merge a working detector

ankushgoel27 replied:
@zricethezav let me know if you would like to add the regex with optional digit added to my PR.

zricethezav replied:
@ankushgoel27, yes please. I'll defer to @4390c336 as he is an engineer at Databricks.

zricethezav commented:
Closing this in favor of @ankushgoel27's PR
